bedita-app/libs/be_thumb.php

```diff
@@ -887,18 +890,28 @@ private function targetFileName() {
      * @return boolean
      */
     private function checkCacheDirectory() {
-        // #769 - avoid file access
-	    if (!file_exists($this->imageInfo['cacheDirectory'])) {
-			if (!mkdir($this->imageInfo['cacheDirectory'], 0777, true)) {
-				$this->triggerError("Error creating cache directory: " . $this->imageInfo['cacheDirectory']);
-				return false;
-			}
-		} elseif (!is_dir($this->imageInfo['cacheDirectory'])) {
-			$this->triggerError("Not a direcotory: " . $this->imageInfo['cacheDirectory']);
-			return false;
-		}
-		return true;
-	}
+        if ($this->cacheDirChecked === true) {
+            return true;
+        }
+
+        if (file_exists($this->imageInfo['cacheDirectory'])) {
+            if (!is_dir($this->imageInfo['cacheDirectory'])) {
+                $this->triggerError($this->imageInfo['cacheDirectory'] . ' is not a directory');
+            
+                return $this->cacheDirChecked = false;
```

Comment:
In this case, subsequent calls to `checkCacheDirectory()` would perform the same checks again and again.

How 'bout using `null` as default value, and then `if ($this->cacheDirChecked !== null) { return $this->cacheDirChecked; }` ?
This might cause issues I haven't thought of, so please evaluate this suggestion _very_ carefully. 😱